docs/reference/modules/cluster/disk_allocator.asciidoc

```diff
@@ -75,13 +75,25 @@ Defaults to `true`. Set to `false` to disable the disk allocation decider. Upon
 Controls the low watermark for disk usage. It defaults to `85%`, meaning that {es} will not allocate shards to nodes that have more than 85% disk used. It can alternatively be set to a ratio value, e.g., `0.85`. It can also be set to an absolute byte value (like `500mb`) to prevent {es} from allocating shards if less than the specified amount of space is available. This setting has no effect on the primary shards of newly-created indices but will prevent their replicas from being allocated.
 // end::cluster-routing-watermark-low-tag[]
 
+`cluster.routing.allocation.disk.watermark.low.max_headroom`::
+(<<dynamic-cluster-setting,Dynamic>>) Controls the max headroom for the low watermark (in case of a percentage/ratio value).
+Defaults to 200GB when `cluster.routing.allocation.disk.watermark.low` is not explicitly set.
+This caps the amount of free space required.
+If set, the `cluster.routing.allocation.disk.watermark.high.max_headroom` needs to be set as well.
```

Comment:
I am not sure which of the specific sentences you refer to :) 

But if it's for the default value of max headroom, I followed the same approach as for the org.elasticsearch.cluster.routing.allocation.DiskThresholdSettings#CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_MAX_HEADROOM_SETTING initialization:

```
            if (CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_WATERMARK_SETTING.exists(settings)) {
                return "-1";
            } else {
                return "20GB";
            }
```

So, if the related watermark is set, then the max headroom setting defaults to -1, which basically means it's ignored during the free bytes calculation of the watermark.

Please unresolve this conversation if this was not sufficient.

Reply:
I am referring to this sentence (the line which the comment is on):

```
If set, the `cluster.routing.allocation.disk.watermark.high.max_headroom` needs to be set as well.
```

This sounds like if `cluster.routing.allocation.disk.watermark.high` is set explicitly, then `cluster.routing.allocation.disk.watermark.high.max_headroom` must always be set. Instead, I think it is ignored (or infinite) if `watermark.high` is set, unless explicitly set.

Reply:
Oh I see the confusion now. The `if set` is referring to the low max headroom setting rather than the high watermark setting. In essence, if someone sets a max headroom value, we say that the higher-level max headroom values need to be set as well. Let me make this more clear by saying `If the low max headroom setting is set...`. Will do similar for the other headrooms of course. Please unresolve this if it's still not clear.